pkg/operator/controller/garden/garden/reconciler.go

```diff
@@ -121,7 +122,19 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return result, nil
 	}
 
-	return reconcile.Result{RequeueAfter: r.Config.Controllers.Garden.SyncPeriod.Duration}, r.updateStatusOperationSuccess(ctx, garden, operationType)
+	if err := r.updateStatusOperationSuccess(ctx, garden, operationType); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// ETCD encryption key rotation requires 2 reconciliations to complete. In prepared phase
+	// the encrypted data has been decrypted and re-encrypted with the new key, but the old key is still present.
+	// The second reconciliation will remove the old key and set the phase to completed.
+	if etcdEncryptionKeyRotationPhase := helper.GetETCDEncryptionKeyRotationPhase(garden.Status.Credentials); etcdEncryptionKeyRotationPhase == gardencorev1beta1.RotationPrepared &&
+		helper.ShouldETCDEncryptionKeyRotationBeAutoCompleteAfterPrepared(garden.Status.Credentials) {
+		return reconcile.Result{RequeueAfter: 1 * time.Nanosecond}, nil
```

Comment:
The wait duration when using the deprecated `Requeue: true` was `5*time.Millisecond`, if the object was never requeued with backoff before that. I'd suggest to use that:
```suggestion
		return reconcile.Result{RequeueAfter: 5 * time.Millisecond}, nil
```